mcs/class/Mono.Data.Tds/Mono.Data.Tds.Protocol/Tds70.cs

```diff
@@ -559,9 +559,8 @@ private void WriteParameterInfo (TdsMetaParameter param)
 			 * If the value is null, not setting the size to 0 will cause varchar
 			 * fields to get inserted as an empty string rather than an null.
 			 */
-			if (colType != TdsColumnType.IntN && colType != TdsColumnType.DateTimeN) {
```

Comment:
Ok, makes sense. Could you please update the commit message with that explanation and also mention it's an extension of the https://github.com/elw00d/mono/commit/7ed9d55b181b0199483734102dc203a1e629e46b fix.